File: in_Python/0682 Baseball Game.py
class Solution(object):
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        st = []
        for i in ops:
            if i not in ['C','+','D']:
                st.append(int(i))
            elif i == 'C':
                st.pop()
            elif i == 'D':
                st.append(st[-1] * 2)
            else:
                st.append(st[-1] + st[-2])
        return sum(st)
        

File: in_Java/0682 Baseball Game.java
class Solution {
    public int calPoints(String[] ops) {
        Stack<Integer> st = new Stack<>();
        int last = 0;
        int last2 = 0;
        for(String s : ops){
            if(s.equals("+")){
                last = st.pop();
                last2 = st.peek();
                st.push(last);
                st.push(last + last2);
            }else if(s.equals("D")){
                st.push(2 * st.peek());
            }else if(s.equals("C")){
                st.pop();
            }else{
                st.push(Integer.parseInt(s));
            }
        }
        int res = 0;
        for(int i : st) res += i;
        return res;
    }
}